src/components/Mission/Mission.jsx

```diff
@@ -0,0 +1,51 @@
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { joinMission, leaveMission } from '../../redux/missions/missions';
+
+const Mission = (props) => {
+  const {
+    id,
+    name,
+    description,
+    join,
+  } = props;
+  const dispatch = useDispatch();
+
+  const handleClick = () => {
+    if (join) {
+      dispatch(leaveMission(id));
+    } else {
+      dispatch(joinMission(id));
+    }
+  };
```

Comment:
"Join Mission" button updates the store successfully by dispatching an action 💪😎 Well done guys ✔